mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -2314,7 +2314,8 @@ OpFoldResult arith::SelectOp::fold(FoldAdaptor adaptor) {
     return trueVal;
 
   // select %x, true, false => %x
-  if (getType().isInteger(1) && matchPattern(adaptor.getTrueValue(), m_One()) &&
+  if (getType().isSignlessInteger(1) &&
+      matchPattern(adaptor.getTrueValue(), m_One()) &&
       matchPattern(adaptor.getFalseValue(), m_Zero()))
     return condition;
```

Comment:
I think we could also have a canon pattern that can broadcast the condition in the vector case, but that's probably better for a separate PR.

Reply:
Since I am not familiar with the Arith + vector use case, I would prefer a separate PR for this.